src/diffusers/utils/hub_utils.py

```diff
@@ -291,9 +291,21 @@ def _get_model_file(
     user_agent: Optional[Union[Dict, str]] = None,
     revision: Optional[str] = None,
     commit_hash: Optional[str] = None,
+    dduf_entries=None,
 ):
     pretrained_model_name_or_path = str(pretrained_model_name_or_path)
-    if os.path.isfile(pretrained_model_name_or_path):
+
+    if dduf_entries:
+        if os.path.join(pretrained_model_name_or_path, weights_name) in dduf_entries:
+            return os.path.join(pretrained_model_name_or_path, weights_name)
+        elif (
+            subfolder is not None
+            and os.path.join(pretrained_model_name_or_path, subfolder, weights_name) in dduf_entries
+        ):
+            return os.path.join(pretrained_model_name_or_path, weights_name)
```

Comment:
same as above, mixing paths on disk and DDUF entries feels weird + can lead to errors on Windows machines

Reply:
fixed !